controllers/ie_controller.go

```diff
@@ -0,0 +1,62 @@
+package controllers
+
+import (
+	"net/http"
+	"reflect"
+
+	mgo "gopkg.in/mgo.v2"
+
+	"github.com/gin-gonic/gin"
+)
+
+// IEController Wrapped for Patient Collection
+type IEController struct {
+	idScope        string
+	db             *mgo.Database
+	collectionName string
+	item           interface{}
+}
+
+// All List All Items
+func (ie *IEController) All(c *gin.Context) {
+	items := ie.itemSlice()
+	ie.getCollection().Find(nil).All(items)
+
+	c.JSON(http.StatusOK, items)
+}
+
+// Create a Patient resource
+func (ie *IEController) Create(c *gin.Context) {
+	err := ie.getCollection().Insert(getJSONBody(c))
+	handleMongoError(c, err)
```

Comment:
This doesn't seem to send a response.  Should it be sending an HTTP 201 (assuming there was not an error)?  Should it send the new URL identifier using the Header `Location` attribute?